securedpreferencestore/src/main/java/devliving/online/securedpreferencestore/EncryptionManager.java

```diff
@@ -107,8 +114,15 @@
 
     SecuredPreferenceStore.KeyStoreRecoveryNotifier mRecoveryHandler;
 
-    public EncryptionManager(Context context, SharedPreferences prefStore, SecuredPreferenceStore.KeyStoreRecoveryNotifier recoveryHandler)
+    public EncryptionManager(Context context, SharedPreferences prefStore, String keyAliasPrefix, SecuredPreferenceStore.KeyStoreRecoveryNotifier recoveryHandler)
```

Comment:
I think key prefix should be marked with `@NonNull` here. That way user will have to distinguishably choose whether they want to provide one (so use this constructor) or not  (use the older constructor with default prefix). It'll prevent developers from using the wrong constructor without explicitly knowing.

Reply:
There are specific checks for null in the code, setting a default filename (same as in the current version). This is to support the same init methods that SecurePreferenceStore class contains.

Reply:
In my opinion we can drop the Old constructor and make this the only constructor but in that case we should mark the `String keyAliasPrefix` as `@Nullable` thus indicating that a `null` value is ok.